windows/deploy_windows_jack.ps1

```diff
@@ -0,0 +1,303 @@
+    "**********************************************************************"
+    ""
+
+    if (-Not (Test-Path -Path $VcVarsBin))
+    {
+        Throw "Microsoft Visual Studio ($BuildArch variant) is not installed. " + `
+            "Please install Visual Studio 2017 or above it before running this script."
+    }
+
+    if (-Not (Test-Path -Path $Env:QtQmakePath))
+    {
+        Throw "The Qt binaries for Microsoft Visual C++ 2017 or above could not be located at $QtMsvcSpecPath. " + `
+            "Please install Qt with support for MSVC 2017 or above before running this script," + `
+            "then call this script with the Qt install location, for example C:\Qt\5.15.2"
+    }
+
+    # Import environment variables set by vcvarsXX.bat into current scope
+    $EnvDump = [System.IO.Path]::GetTempFileName()
+    Invoke-Native-Command -Command "cmd" `
+        -Arguments ("/c", "`"$VcVarsBin`" && set > `"$EnvDump`"")
+
+    foreach ($_ in Get-Content -Path $EnvDump)
+    {
+        if ($_ -Match "^([^=]+)=(.*)$")
+        {
+            Set-Item "Env:$($Matches[1])" $Matches[2]
+        }
+    }
+
+    Remove-Item -Path $EnvDump -Force
+}
+
+# Build Jamulus x86_64 and x86
+Function Build-App
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $BuildConfig,
+        [Parameter(Mandatory=$true)]
+        [string] $BuildArch
+    )
+
+    Invoke-Native-Command -Command "$Env:QtQmakePath" `
+        -Arguments ("$RootPath\$AppName.pro", "CONFIG+=$BuildConfig $BuildArch $BuildOption", `
+        "-o", "$BuildPath\Makefile")
+
+    Set-Location -Path $BuildPath
+    Invoke-Native-Command -Command "nmake" -Arguments ("$BuildConfig")
+    Invoke-Native-Command -Command "$Env:QtWinDeployPath" `
+        -Arguments ("--$BuildConfig", "--compiler-runtime", "--dir=$DeployPath\$BuildArch",
+        "$BuildPath\$BuildConfig\$AppName.exe")
+
+    Move-Item -Path "$BuildPath\$BuildConfig\$AppName.exe" -Destination "$DeployPath\$BuildArch" -Force
+    Invoke-Native-Command -Command "nmake" -Arguments ("clean")
+    Set-Location -Path $RootPath
+}
+
+# Build and deploy Jamulus 64bit and 32bit variants
+function Build-App-Variants
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $QtInstallPath
+    )
+
+    foreach ($_ in ("x86_64", "x86"))
+    {
+        $OriginalEnv = Get-ChildItem Env:
+        Initialize-Build-Environment -QtInstallPath $QtInstallPath -BuildArch $_
+        Build-App -BuildConfig "release" -BuildArch $_
+        $OriginalEnv | % { Set-Item "Env:$($_.Name)" $_.Value }
+    }
+}
+
+# Build Windows installer
+Function Build-Installer
+{
+    foreach ($_ in Get-Content -Path "$RootPath\$AppName.pro")
+    {
+        if ($_ -Match "^VERSION *= *(.*)$")
+        {
+            $AppVersion = $Matches[1]
+            break
+        }
+    }
+
+    Invoke-Native-Command -Command "$WindowsPath\NSIS\makensis" `
+        -Arguments ("/v4", "/DAPP_NAME=$AppName", "/DAPP_VERSION=$AppVersion", `
```

Comment:
You should be able to pass an argument to the nsis system here

Reply:
@ann0see  Do you mean also modify the installer.nsi to handle both? Or can that be separate?
Keeping the installer_jack.nsi sounds a lot cleaner to me.

Reply:
If we have two installers that also means two places to maintain. The we must weigh how much "bloat" would be in one installer if we use one for both.

Reply:
My guess is that it will double the size when using a single installeer and the build process needs to altered for this as the "temp" deploy folders x86 and x86_64 will need to be different. Also I thought the sort of agreement was already to keep them seperate as you don't want to confuse the majority of the users. JACK adds a lot more complexticity for standard users.

Reply:
I don't think it is a good Idea to have a single installer for now.
Because, for now, The use case is:
- You dont want to use JACK on Windows -> Install regular Jamulus and enjoy
- You want to use JACK on Windows -> you have to use a custom build of Jamulus (linux sound c++ build on windows).

Reply:
Ok. But if that’s the case, we could go for a separate file. Waiting for others to comment

Reply:
Most PC JACK users a very techy, in my view.  (I'd say the same for most 32bit PC users, too, with some bit of kit they just can't leave behind.)

In fact, I'd fall on the side that offering JACK to everyone on Windows would cause more problems than it solves.  I'd be pretty firmly on the side of a separate installer.

Reply:
Ok. So then I think we agreed on a delegate installer. Probably giving a hint that this installs the JACK version would be good (same as the ASIO warning but without checks for now).

Reply:
> hint that this installs the JACK version

Excellent suggestion I would say, will look into updating the installer_jack.nsi and a this dialog.

Reply:
Adding this dialog means adding text for this. The dialog will not have the button to link to the jamulus.io site for the installation of JACK but just text to indicate it needs JACK.

At the same time, this allows the additional check for "Is JACK installed?" option?
The check is very similar as with ASIO. Shall I just add this as well?

Reply:
> At the same time, this allows the additional check for "Is JACK installed?" option?

Yes. You could just check for jack.exe, I suppose?

Reply:
For ASIO the existence of a registry is checked, JACK has a similar registry key (when it is installed). So same logic can be used. Will add this to the installer for jack.

Reply:
Ah ok. That's great!